txclib/commands.py

```diff
@@ -605,3 +607,28 @@ def _set_project_option(resource, name, value, path_to_tx, func_name):
     prj = project.Project(path_to_tx)
     getattr(prj, func_name)(resources, value)
     prj.save()
+
+
+def get_branch_from_options(options, project_root):
+    """ Returns the branch name that needs to be used in command
+    based on parser options.
+
+    options: optparse parser options as returned from `parse()`
+    project_root: project root directory
+    """
+    if not options.branch:
+        if options.branchname:
```

Comment:
I suggest moving the `if options.branchname` one level up, and merge it with the if statement that follows.
```
if options.branchname:
    if not options.branch:
        ...
        sys.exit(1)
    return ...
```